Detections/SecurityEvent/PotentialKerberoast.yaml

```diff
@@ -41,7 +41,25 @@ query: |
   | where ServiceName !contains "$" and ServiceName !contains "krbtgt" 
   | parse EventData with * 'TargetUserName">' TargetUserName "<" *
   | where TargetUserName !contains "$@" and TargetUserName !contains ServiceName
-  | parse EventData with * 'IpAddress">::ffff:' ClientIPAddress "<" *;
+  | parse EventData with * 'IpAddress">::ffff:' ClientIPAddress "<" *
+  ),
+  (
+  WindowsEvent
+  | where TimeGenerated >= ago(starttime)
+  | where EventID == 4769 and EventData has '0x17' and EventData has '0x40810000' and EventData has 'krbtgt'
+  | extend TicketEncryptionType = tostring(EventData.TicketEncryptionType)
+  | where TicketEncryptionType == '0x17'
+  | extend TicketOptions = tostring(EventData.TicketOptions)
+  | where TicketOptions == '0x40810000'
+  | extend Status = tostring(EventData.Status)
+  | where Status == '0x0'
+  | extend ServiceName = tostring(EventData.ServiceName)
+  | where ServiceName !contains "$" and ServiceName !contains "krbtgt"
+  | extend TargetUserName = tostring(EventData.TargetUserName) 
+  | where TargetUserName !contains "$@" and TargetUserName !contains ServiceName
+  | extend TargetUserName = tostring(EventData.TargetUserName) 
+  | extend ClientIPAddress = tostring(EventData.IpAddress) 
+  ));
   let Kerbevent23h = Kerbevent
```

Comment:
An exam type question: why is this very inefficient and how can it be fixed? Don't start changing, just food for thought.